app/uk/gov/hmrc/tradergoodsprofilesrouter/connectors/BaseConnector.scala

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright 2023 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.tradergoodsprofilesrouter.connectors
+
+import play.api.http.Status.OK
+import play.api.libs.json.{JsResult, Reads}
+import uk.gov.hmrc.http.HttpReads.Implicits._
+import uk.gov.hmrc.http.{HttpErrorFunctions, HttpResponse, UpstreamErrorResponse}
+import uk.gov.hmrc.http.client.RequestBuilder
+
+import scala.concurrent.{ExecutionContext, Future}
+
+trait BaseConnector extends HttpErrorFunctions {
+
+  implicit class HttpResponseHelpers(response: HttpResponse) {
+
+    def as[A](implicit reads: Reads[A]): Future[A] =
+      response.json
+        .validate[A]
+        .map(result => Future.successful(result))
+        .recoverTotal(error => Future.failed(JsResult.Exception(error)))
+
+    def error[A]: Future[A] =
+      Future.failed(UpstreamErrorResponse(response.body, response.status))
+
+  }
+
+  implicit class RequestBuilderHelpers(requestBuilder: RequestBuilder) {
+    def executeAndDeserialise[T](implicit ec: ExecutionContext, reads: Reads[T]): Future[T] =
+      requestBuilder
+        .execute[HttpResponse]
+        .flatMap { response =>
+          response.status match {
+            case OK => response.as[T]
+            case _  =>
+              response.error
+          }
+        }
+
+    /**
+      * This method will be used for other endpoint e.g. delete
+      */
+    def executeAndExpect(expected: Int)(implicit ec: ExecutionContext): Future[Unit] =
```

Comment:
I have kept it as it will be used for delete end point, we may again look for code if response is blank with just status code. I have added the comment on top of the method, but is it worth commenting the whole code or removing it?